pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -51,10 +52,16 @@ type PipelineRunState []*ResolvedPipelineTask
 // and scheduling the next tasks. It is a collection of list of ResolvedPipelineTask, graph of DAG tasks, graph of
 // finally tasks, cache of skipped tasks.
 type PipelineRunFacts struct {
-	State           PipelineRunState
-	SpecStatus      v1beta1.PipelineRunSpecStatus
-	TasksGraph      *dag.Graph
-	FinalTasksGraph *dag.Graph
+	State            PipelineRunState
+	SpecStatus       v1beta1.PipelineRunSpecStatus
+	TasksGraph       *dag.Graph
+	FinalTasksGraph  *dag.Graph
+	StartTime        *time.Time
+	FinallyStartTime *time.Time
+	PipelineTimeout  *time.Duration
+	TasksTimeout     *time.Duration
+	FinallyTimeout   *time.Duration
+	Clock            clock.PassiveClock
```

Comment:
The most direct way to get this information available to the new functions on `ResolvedPipelineTask` was to add them to `PipelineRunFacts` - but I feel like it's probably not the _optimal_ way to do it. This is just a first draft on getting rid of `getFinallyTaskRunTimeout` and `getTaskRunTimeout` completely, so I am completely open to alternative approaches for getting this information available to `skipBecausePipelineRunPipelineTimeoutReached`, `skipBecausePipelineRunTasksTimeoutReached`, and `skipBecausePipelineRunFinallyTimeoutReached`.

Reply:
I suggest that we group the timeouts state fields that are referenced in `PipelineRunFacts`, such as is done for `State`, what do you think @abayer? 

For example:

```go

type PipelineRunFacts struct {
	State            PipelineRunState
	SpecStatus       v1beta1.PipelineRunSpecStatus
	TasksGraph       *dag.Graph
	FinalTasksGraph  *dag.Graph
        TimeoutsState       PipelineRunTimeoutsState
}

type PipelineRunTimeoutsState struct {
	StartTime        *time.Time
	FinallyStartTime *time.Time
	PipelineTimeout  *time.Duration
	TasksTimeout     *time.Duration
	FinallyTimeout   *time.Duration
	Clock            clock.PassiveClock
}
```

I think it makes the code cleaner, and easier to pass around the timeouts state from facts to other functions, instead of passing `PipelineRunFacts` as a whole or the list of timeouts fields as arguments

Reply:
👍